training/src/main/java/wolox/training/services/OpenLibraryService.java

```diff
@@ -0,0 +1,104 @@
+package wolox.training.services;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.springframework.stereotype.Service;
+import wolox.training.exceptions.BookNotFoundException;
+import wolox.training.exceptions.ConnectionFailedException;
+import wolox.training.exceptions.CouldNotReadBookFromAPI;
+import wolox.training.models.BookDTO;
+
+@Service
+public class OpenLibraryService {
+
+    public BookDTO bookInfo(String isbn)
+        throws IOException, JSONException, ConnectionFailedException, BookNotFoundException, CouldNotReadBookFromAPI {
+
+        BookDTO bookDTO;
+
+        URL url = new URL("https://openlibrary.org/api/books?bibkeys=ISBN:"+isbn+"&format=json&jscmd=data");
+        HttpURLConnection con = (HttpURLConnection) url.openConnection();
+        con.setRequestMethod("GET");
+
+        if(con.getResponseCode()!=HttpURLConnection.HTTP_OK){
+            throw new ConnectionFailedException(con.getResponseCode(), con.getResponseMessage());
+        }
+
+        String response = getResponseString(con.getInputStream());
+
+        JSONObject jo = new JSONObject(response);
+
+        if(jo.length()<1) {
```

Comment:
Just wondering if this is the proper validation, I searched a book by a random invalid isbn using the API and it returns an empty JSON ( "{}, string length = 2 ). Maybe you could check if the return value of .getNames() is empty instead?

Reply:
I've just debugged it with an invalid isbn an the lenght is 0, I think this might be because its a JSONObject, not a string. When I did this I didn't like to compare by length, but I did not find another way.

Reply:
And how about checking for the value of JSONObject.getNames(jo) instead?

Reply:
Resolved ir with _jo.keys().hasNext()_